--- origsrc/ocurl/Makefile.in	2009-11-07 15:05:52.000000000 -0600
+++ src/ocurl/Makefile.in	2010-02-14 23:22:27.934080500 -0600
@@ -28,7 +28,7 @@ CURLOBJS	= curl.cmx
 CURLBCOBJS	= $(CURLOBJS:.cmx=.cmo)
 
 CURLFLAGS	= -ccopt @CURLFLAGS@
-CURLCLIBS	= -cclib -lcurl-helper -cclib "@CURLLIBS@"
+CURLCLIBS	= -cclib -lcurl-helper -cclib "@CURLFLAGS@ @CURLLIBS@"
 
 ifeq (@OCAMLBEST@,opt)
 TARGETS		= curl.cma curl.cmxa libcurl-helper.a
@@ -46,7 +46,7 @@ examples:
 		(cd examples; $(MAKE))
 
 curl.cma:	$(CURLBCOBJS) libcurl-helper.a
-		$(OCBYTE) -custom -a $(FLAGS) $(CURLFLAGS) -o $@ $(CURLBCOBJS) $(CURLCLIBS)
+		$(OCBYTE) -custom -a $(FLAGS) -dllib dllcurl-helper $(CURLFLAGS) -o $@ $(CURLBCOBJS) $(CURLCLIBS)
 
 curl.cmxa:	$(CURLOBJS) libcurl-helper.a
 		$(OCOPT) -a $(FLAGS) $(CURLFLAGS) $(LIBS) $(CURLOBJS) -o $@ $(CURLCLIBS)
@@ -60,16 +60,16 @@ curl.cmxa:	$(CURLOBJS) libcurl-helper.a
 .ml.cmo:
 		$(OCBYTE) -c $(FLAGS) $< -o $@
 
-libcurl-helper.a:	$(CURLHELPEROBJS)
-		ar crus libcurl-helper.a $(CURLHELPEROBJS)
-		ranlib libcurl-helper.a
+libcurl-helper.a dllcurl-helper.so:	$(CURLHELPEROBJS)
+		ocamlmklib -o curl-helper $(CURLHELPEROBJS) @CURLFLAGS@
 
 .c.o:
 		$(CC) -c -Wall $(CFLAGS) $(CAMLINCDIR) $< -o $@
 
 install:
 ifeq ($(FINDLIB),ocamlfind)
-		ocamlfind install curl META $(wildcard *.cmi) \
+		ocamlfind install curl META $(wildcard *.so) $(wildcard *.cmi) \
+			 $(wildcard *.cmx) $(wildcard *.mli) \
 			 $(wildcard *.a) $(wildcard *.cma) $(wildcard *.cmxa)
 else
 		mkdir -p $(INSTALLDIR)
