--- origsrc/ocurl-0.6.0/Makefile.in	2013-10-28 21:16:31.692848000 -0500
+++ src/ocurl-0.6.0/Makefile.in	2013-10-28 21:54:14.223257400 -0500
@@ -42,10 +42,10 @@ targets:	$(TARGETS) examples
 examples:
 		(cd examples; $(MAKE))
 
-curl.cma:	$(CURLBCOBJS) dllcurl-helper.so
+curl.cma:	$(CURLBCOBJS) libcurl-helper.a
 		$(OCAMLMKLIB) -o curl $(CURLBCOBJS) -oc curl-helper $(CLIBS)
 
-curl.cmxa:	$(CURLOBJS) dllcurl-helper.so
+curl.cmxa:	$(CURLOBJS) libcurl-helper.a
 		$(OCAMLMKLIB) -o curl $(CURLOBJS) -oc curl-helper $(CLIBS)
 
 .ml.cmx:
